### PR TITLE
Revisions in Server to avoid application fail on Server errors

### PR DESCRIPTION
**Pull request by MikePooh**
Most of qFatals and exit()s has been replaced by handlers which emits errorOccured signal. This behavior will prevent qApplication fail on Server errors.

### CODE REVIEW

File: server/server.cpp [hunk 1]
```diff
@@ -58,18 +58,20 @@ using namespace Cutelyst;
...
-        std::cout << "Installing EPoll event loop" << std::endl;
-        QCoreApplication::setEventDispatcher(new EventDispatcherEPoll);
-    }
-#endif
+//#ifdef Q_OS_LINUX
+//    if (!qEnvironmentVariableIsSet("CUTELYST_QT_EVENT_LOOP")) {
+//        std::cout << "Installing EPoll event loop" << std::endl;
+//        QCoreApplication::setEventDispatcher(new EventDispatcherEPoll);
+//    }
+//#endif
```
Comment (dantti) [inline server/server.cpp:74]: If this change is going to happen this comments can go

File: server/server.cpp [hunk 2]
```diff
@@ -513,7 +515,12 @@ int Server::exec(Cutelyst::Application *app)
     Q_D(Server);
     std::cout << "Cutelyst-WSGI starting" << std::endl;
 
-
+#ifdef Q_OS_LINUX
+    if (!d->userEventLoop) { //launchs Event loop only in case standalone app, not embedded
```
Comment (dantti) [inline server/server.cpp:519]: This is missing qEnvironmentVariableIsSet("CUTELYST_QT_EVENT_LOOP") as cutelystd might be used with default event loop.
The comment is also a bid misleading, as I might embedded it to get better at LTO and still want EPool.

**Review by dantti — CHANGES_REQUESTED** [review]
Mostly ok, thanks!

**Comment by dantti** [comment]
while you are at it #213 is related and might hit you :)

File: server/server.cpp [hunk 3]
```diff
@@ -58,18 +58,20 @@ using namespace Cutelyst;
 Server::Server(QObject *parent) : QObject(parent),
     d_ptr(new ServerPrivate(this))
 {
+    std::cout << "Mikes Version of Cutelyst" << std::endl;
```
[the remark this answers is not in the document]
Comment (MikePooh) [reply server/server.cpp:61]: Haha, indeed. I just placed it to be sure I am using forked version of library, not original on tests :)

File: server/main.cpp [hunk 4]
```diff
@@ -33,6 +33,9 @@ int main(int argc, char *argv[])
 
     Cutelyst::Server server;
 
+    QObject::connect(&server, &Cutelyst::Server::errorOccured, [](const QString &error){
+        qFatal("Server terminated due to error %s", error.toStdString().c_str());});
```
Comment (dantti) [inline server/main.cpp:37]: better use qPrintable

**Pull request merged**